### PR TITLE
Add PostAttestationV2 API

### PR DESCRIPTION
**Pull request by mehdi-aouadi**
## PR Description
- Add PostAttestationV2 API
- Handle arrays with different item types deserialisation
- Add a header based schema selector

## Fixed Issue(s)

#8424 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/MilestoneDependentTypesUtil.java [hunk 1]
```diff
@@ -93,11 +96,28 @@ public static <T extends SszData> DeserializableTypeDefinition<? extends T> slot
         .getJsonTypeDefinition();
   }
 
+  public static <T extends SszData> DeserializableTypeDefinition<? extends T> headerBasedSelector(
+      final Map<String, String> headers,
+      final SchemaDefinitionCache schemaDefinitionCache,
+      final Function<SchemaDefinitions, SszSchema<? extends T>> getSchema) {
+    try {
+      final SpecMilestone milestone = SpecMilestone.forName(headers.get(HEADER_CONSENSUS_VERSION));
```
[the remark this answers is not in the document]
Comment (mehdi-aouadi) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/MilestoneDependentTypesUtil.java:104]: There is already a try catch

File: infrastructure/json/src/main/java/tech/pegasys/teku/infrastructure/json/types/OpenApiTypeDefinition.java [hunk 2]
```diff
@@ -43,6 +43,10 @@ default boolean isEquivalentToDeserializableType(final DeserializableTypeDefinit
     return this == type;
   }
 
+  default boolean isEquivalentToDeserializableType(final SerializableTypeDefinition<?> type) {
```
Comment (gfukushima) [inline infrastructure/json/src/main/java/tech/pegasys/teku/infrastructure/json/types/OpenApiTypeDefinition.java]: Just double checking... was this meant to be named `isEquivalentToSerializableType`?

File: data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v2/beacon/PostAttestationsV2IntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+            dataStructureUtil.randomAttestation(),
+            dataStructureUtil.randomAttestation(),
+            dataStructureUtil.randomAttestation());
+
+    final Response response =
+        post(
+            PostAttestationsV2.ROUTE,
+            JsonUtil.serialize(
+                attestations,
+                SerializableTypeDefinition.listOf(
```
Comment (gfukushima) [inline data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v2/beacon/PostAttestationsV2IntegrationTest.java]: nit: this is reused in all methods and could probably get extracted into a private method

**Review by gfukushima — APPROVED** [review]
lgtm with some minor asks

File: infrastructure/json/src/main/java/tech/pegasys/teku/infrastructure/json/types/OpenApiTypeDefinition.java (hunk 2 above)
Comment (mehdi-aouadi) [reply infrastructure/json/src/main/java/tech/pegasys/teku/infrastructure/json/types/OpenApiTypeDefinition.java]: My bad, this isn't used anymore anyway. Removed it

File: data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v2/beacon/PostAttestationsV2IntegrationTest.java (hunk 3 above)
Comment (mehdi-aouadi) [reply data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v2/beacon/PostAttestationsV2IntegrationTest.java]: Good catch, refactored it

**Pull request merged**